go/ngql/types.go

```diff
@@ -330,9 +341,32 @@ func getArgValue(arg interface{}, nomsType *types.Type) (types.Value, error) {
 	return nil, fmt.Errorf("%s is not a subtype of %s", nomsVal.Type().Describe(), nomsType.Describe())
 }
 
-func getCollectionArgs(typ *types.Type, len uint64, args map[string]interface{}, factory iteratorFactory) (iter interface{}, nomsKey, nomsThrough types.Value, count uint64, singleExactMatch bool, err error) {
+type nomsCollection interface {
+	types.Value
+	Len() uint64
+}
+
+func getCollectionArgs(col nomsCollection, args map[string]interface{}, factory iteratorFactory) (iter interface{}, nomsKey, nomsThrough types.Value, count uint64, singleExactMatch bool, err error) {
```

Comment:
I think "using keys prevents using any other arguments" makes sense to prevent key/count, but as a general rule, it seems wrong?